internal/controllers/topology/cluster/scope/state_test.go

```diff
@@ -115,7 +115,93 @@ func TestIsUpgrading(t *testing.T) {
 	}
 }
 
-func TestUpgrading(t *testing.T) {
+func TestMPIsUpgrading(t *testing.T) {
+	g := NewWithT(t)
+	scheme := runtime.NewScheme()
+	g.Expect(clusterv1.AddToScheme(scheme)).To(Succeed())
```

Comment:
You will need to add `expv1` to the scheme as well since `MachinePools` are part of that package instead of `clusterv1`.
```
g.Expect(expv1.AddToScheme(scheme)).To(Succeed())
```